Various fixes

Dimoks opened the pull request.

Dimoks committed:
Adding a ring buffer to mzx compression

Dimoks committed:
Fix unfix_alpha?

Dimoks committed:
Reading palette from png.

Dimoks committed:
Fix rgb565?

Dimoks committed:
Creating new mzp archive

loicfrance commented on mzp.py:
```diff
@@ -227,6 +233,13 @@ def update_offsets(self, alignment: int = MZP_DEFAULT_ALIGNMENT) :
             if alignment > 0 :
                 offset += alignment - offset % alignment
     
+    def add_entry(self, data: bytes, alignment: int = MZP_DEFAULT_ALIGNMENT) -> MzpArchiveEntry:
```
why specify alignment here ? alignement must be the same for all entries, and is only used when writing mzp

loicfrance commented on mzp.py:
```diff
@@ -49,6 +49,8 @@ def fix_alpha(a: np.uint8) :
 def unfix_alpha(a: np.uint8) :
     if a == 0xFF :
         return a
+    elif a == 0 :
```
This commit was labeled "Fix unfix_alpha?". However, I don't think the result of this function would be any different from before. It might have a small impact on performance though (positive or negative, I haven't measured).

loicfrance commented on mzp.py:
```diff
@@ -438,8 +463,8 @@ def img_write(self, dest: None = None) -> BytesIO: ...
         crop = self.tile_crop
-        height = self.height - self.tile_y_count * crop * 2
-        width = self.width - self.tile_x_count * crop * 2
+        height = self.height
+        width = self.width
```
What is the purpose of this change ? Wouldn't this add unused pixels to the image ?

loicfrance committed:
[MZX] move ring buffer search to method, use RB in lvl 3

Avoids calling `ring_buf.get()` for each index which costs a `_file.tell()` and two `_file.seek()` everytime

+ use `invert` to create the XOR'd value instead of multiplying it every time.

Dimoks committed:
Revert "Fix unfix_alpha?"
